File: lectures/code/kr_04_02.c
#include <stdio.h>
#include <string.h>

void zap(char y[]) {
    printf("Y start  zap: %s\n",y);
    strcpy(y, "CHANGED");
    printf("Y end    zap: %s\n",y);
}

int main(){
    char x[] = "ORIGINAL";
    printf("X before zap: %s\n",x);
    zap(x);
    printf("X after  zap: %s\n",x);
}

File: lectures/code/kr_04_02.py
def zap(y):
    print('Y start  zap:',y)
    y = 'CHANGED'
    print('Y end    zap:',y)

x = 'ORIGINAL'
print('X before zap:',x)
zap(x)
print('X after  zap:',x)
